src/main/java/driver/Duke.java

```diff
@@ -0,0 +1,89 @@
+package driver;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Scanner;
+import task.TaskList;
+import command.HelloCommand;
+import command.Command;
+import command.TextFormatter;
+import command.ByeCommand;
+
+/**
+ *
+ */
+
+public class Duke {
+
+    Ui myUserInterface;
+    TaskList myTaskList;
+    Storage myTaskManager;
+    Parser myParser;
+
+    /**
+     *
+     */
+
+    public Duke(String x) {
+        myUserInterface = new Ui();
+        //Create TaskList for Duke
+        myTaskList = new TaskList();
+        myTaskManager = new Storage(x);
+        myParser = new Parser();
+        //Copy tasks from .txt file to my taskList
+        try {
+            myTaskManager.loadTasks(myTaskList);
+        }
+        catch (FileNotFoundException err) {
+            System.out.println(err);
+        }
+    }
+
+    /**
+     *
+     */
+
+    public void run()  {
+        //Say Hello
+        HelloCommand sayHello = new HelloCommand();
+        sayHello.executeCommand(myTaskList,myUserInterface);
+
+        Scanner userInput = new Scanner(System.in);
+        while(userInput.hasNextLine()) {
```

Comment:
This works well to handle user input! Have you considered using the Scanner class's scanner.hasNext() method (userinput = bye) to see when to end reading from the scanner though?